client/landing/stepper/declarative-flow/internals/types.ts

```diff
@@ -139,7 +140,14 @@ export type Flow = {
 	 * A hook that is called in the flow's root at every render. You can use this hook to setup side-effects, call other hooks, etc..
 	 */
 	useSideEffect?: UseSideEffectHook< ReturnType< Flow[ 'useSteps' ] > >;
-
+	/**
+	 * Used for overriding the props recorded by the default/framework-handled Tracks event recorders.
+	 * Can pass any properties that should be recorded for the event.
+	 *   - Currently only applicable to step-navigation events.
+	 */
+	useTracksEventProps?: (
+		event: StepperTracksEventStepNavigation
+	) => Record< string, string | number | null > | undefined;
```

Comment:
WDYT about making this middleware style and giving it access to the whole event to intercept. Maybe some flows want to change the event name or whatever?

```js
useTracksEventProps?: (
		event: Readonly<StepperTracksEventStepNavigation> // make Readonly to prevent mutation.
) => StepperTracksEventStepNavigation & Record<string, string | number | boolean > | undefined;
```


Then the flows would do

```js
useTracksEventProps( event ) {
	if( event.name === 'some-event-i-want-to-change' ) {
		return { ...event, 	'my-custom-prop': prop };
	}
}
```

This would make for a better API and lowers the chances of people modifying Stepper.

Reply:
That doesn't sound bad. So they can change the event that gets recorded alltogether.

I'm curious though if we have a current use case. I do get the feeling of YAGNI all around this tbh (including current)

Reply:
> That doesn't sound bad

aka sounds good 😁 

Not meant my comment as blocker. We'll find the use I'm sure. A lot flows do log custom events, so will dig this out a bit more.

Reply:
Sounds good. I'm not fond of dead abstractions/functionality, but it's not that severe.

I think there's a bit of refactoring underneath to have this form of polymorphic behavior (to both add/override current event props, and send a different Tracks event to record). 

Let's take this in a follow-up. I sense a couple of nuances propping up / not blocking. Outside of some cleanup in the event names, is there anything else flashing out? Were you able to test `connect-domain` in other PR?

Reply:
Sounds good. I didn't test the other PR in case we changed things here. I'll check.

Reply:
Reviewed